core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -95,28 +93,70 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
         partition.key, partition.specId, partition.dataRecordCount, partition.dataFileCount);
   }
 
-  private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
-    CloseableIterable<FileScanTask> tasks = planFiles(scan);
+  @VisibleForTesting
```

Comment:
Could we consolidate these two VisibleForTesting methods?  Would just exposing dataFiles() work for those tests?  I don't really exposing Partition inner class.